lib/engines/content_object_store/app/controllers/content_object_store/content_block/editions_controller.rb

```diff
@@ -74,17 +77,37 @@ def review_links
     end
   end
 
+  def schedule_publishing
+    @content_block_document = @content_block_edition.document
+    @edition_params = edition_params
+
+    new_edition = ContentObjectStore::ContentBlock::Edition.new(@edition_params)
+    new_edition.document.id = @content_block_document.id
+
+    render :schedule_publishing
+  end
+
   def update
     content_block_edition = ContentObjectStore::ContentBlock::Edition.find(params[:id])
     @schema = ContentObjectStore::ContentBlock::Schema.find_by_block_type(content_block_edition.document.block_type)
 
-    new_content_block_edition = ContentObjectStore::UpdateEditionService.new(
-      @schema,
-      content_block_edition,
-    ).call(edition_params)
+    new_edition = edition_params
+    if params[:schedule_publishing] == "schedule"
+      new_edition = edition_params.merge!(scheduled_publication_params)
```

Comment:
this gets moved to `workflow_controller` in subsequent commit, and will need to be refactored to handle errors on the scheduling form